inst/chunks/fun_normal_time_se.stan

```diff
@@ -0,0 +1,106 @@
+  /* multi-normal log-PDF for time-series covariance structures
+   * in Cholesky parameterization and assuming homogoneous variances
+   * and known standard errors
+   * Args:
+   *   y: response vector
+   *   mu: mean parameter vector
+   *   sigma: residual standard deviation
+   *   se2: square of user defined standard errors
+   *   chol_cor: cholesky factor of the correlation matrix
+   *   nobs: number of observations in each group
+   *   begin: the first observation in each group
+   *   end: the last observation in each group
+   * Returns:
+   *   sum of the log-PDF values of all observations
+   */
+  real normal_time_hom_se_lpdf(vector y, vector mu, real sigma, data vector se2,
+                               matrix chol_cor, int[] nobs, int[] begin, int[] end) {
+    int I = size(nobs);
+    vector[I] lp;
+    for (i in 1:I) {
+      matrix[nobs[i], nobs[i]] L;
+      L = sigma * chol_cor[1:nobs[i], 1:nobs[i]];
+      // need to add 'se' to the covariance matrix itself
+      L = multiply_lower_tri_self_transpose(L);
```

Comment:
In this homogeneous case you can pull the sigma * L and the multiply_lower_tri_self_transpose out of the loop. This is always the same. Just subset the respective covariance matrix for each row to what is observed is what is needed per data group.